src/modules/companies/companies.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { ApiNotFoundResponse, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+import { CompaniesService } from './companies.service';
+import { GetCompanyByRegistrationNumberQuery } from './dto/get-company-by-registration-number-query.dto';
+import { GetCompanyResponse } from './dto/get-company-response.dto';
+
+@ApiTags('companies')
+@Controller('companies')
+export class CompaniesController {
+  constructor(private readonly companiesService: CompaniesService) {}
+
+  @Get()
+  @ApiOperation({
+    summary: 'Get company by Companies House registration number.',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Returns the company',
+    type: GetCompanyResponse,
+  })
+  @ApiNotFoundResponse({
+    description: 'Company not found.',
+  })
+  getCompanyByRegistrationNumber(@Query() query: GetCompanyByRegistrationNumberQuery): Promise<GetCompanyResponse> {
```

Comment:
As far as I'm aware, Typescript does not have this feature (if we throw rather than return an exception): https://www.learningtypescript.com/articles/why-typescript-doesnt-include-a-throws-keyword

Reply:
Same as above please.

Reply:
From my understanding, using a union type would be appropriate if the function *returned* those errors, but instead it throws them. See the Union Types section in https://www.learningtypescript.com/articles/why-typescript-doesnt-include-a-throws-keyword

Reply:
Understood, in this case should we let the code handle the Error being thrown instead of returning the value? Perhaps what the article has suggested?

```typescript
if (value instanceof Error) {
  console.error("Not ready to get value:", value);
} else {
  console.log("Got a value:", value);
}
```

Reply:
I'm not sure that there would be any benefit to doing so in this case. NestJS already handles known thrown exceptions (e.g., `NotFoundException`) with appropriate status codes by default, and throwing unhandled errors isn't a concern here as NestJS handles these with a `500` response by default.